src/portfft/common/host_fft.hpp

```diff
@@ -0,0 +1,53 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's portFFT
+ *
+ **************************************************************************/
+
+#ifndef PORTFFT_COMMON_HOST_FFT_HPP
+#define PORTFFT_COMMON_HOST_FFT_HPP
+
+#include "portfft/defines.hpp"
+#include <complex>
+
+namespace portfft {
+namespace detail {
+
+/**
+ * Host Naive DFT. Works OOP only
+ * @tparam T Scalar Type
+ * @param input input pointer
+ * @param output output pointer
+ * @param fft_size fft size
+ */
+template <typename T>
+void naive_dft(std::complex<T>* input, std::complex<T>* output, IdxGlobal fft_size) {
```

Comment:
How big a DFT do we expect to compute with this? Can we use portFFT's SYCL implementation here?

Reply:
any size FFTs are acceptable, any reason you ask that question ?. In case you are worried about the time to compute, I can replace this with a CT.
We cannot use our kernels here it would require jitting of a new kernel, at that point a host side CT ( which I assume this will become eventually) would be faster.

Reply:
With the scaling of naive DFTs, we should be careful that calling this function does not become more expensive than the actual DFT we are trying to compute. 

I also don't especially like the notion that we'd need a host implementation of CT - we're trying to write a GPU implementation, not a CPU implementation!